wsmaster/che-core-api-installer/src/main/java/org/eclipse/che/api/installer/server/InstallerRegistry.java

```diff
@@ -28,12 +30,47 @@
 public interface InstallerRegistry {
 
     /**
+     * Adds installer to the registry.
+     *
+     * @param installer
+     *      the installer to add
+     * @throws InstallerConflictException
+     *         if installer with corresponding {@link InstallerFqn} already exists
+     * @throws InstallerException
+     *         if unexpected error occurred
+     */
+    void add(Installer installer) throws InstallerException;
+
+    /**
+     * Updates installer in the registry.
+     *
+     * @param installer
+     *      the installer to update
+     * @throws InstallerNotFoundException
+     *         if installer with corresponding {@link InstallerFqn} does not exist in the registry
+     * @throws InstallerException
+     *         if unexpected error occurred
+     */
+    void update(Installer installer) throws InstallerException;
+
+    /**
+     * Removes installer from the registry.
+     *
+     * @param fqn
+     *      the installer fully-qualified key
+     *      {@link InstallerFqn}
+     * @throws InstallerException
+     *         if unexpected error occurred
+     */
+    void remove(InstallerFqn fqn) throws InstallerException;
```

Comment:
I think you should not use `InstallerFqn` object here, and use string key value like `getOrderedInstallers` or `getInstaller` methods do. Or you should rework all methods to use InstallerFqn. I prefer to use string value.